src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -735,5 +735,14 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Info, // Only used for the suggested edit
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor GenericGeneratorError = new DiagnosticDescriptor(
+			id: "D2L0095",
+			title: "A source generator encountered an unexpected situation. This is a bug in D2L.CodeStyle.Analyzers.",
+			messageFormat: "The source generator {0} encountered an unexpected situation: {1}",
+			category: "Generators",
+			defaultSeverity: DiagnosticSeverity.Error,
+			isEnabledByDefault: true
+		);

```

Comment:
For places where I'm making assumptions about C# trees that I believe are true, even accounting for like partially written code in VS (but it is easy to be wrong about that)